src/index.ts

```diff
@@ -66,8 +84,11 @@ export const getEq = <A extends Sum.AnyMember>(eqs: Eqs<A>): Eq<A> =>
 
     const eq = eqs[xk as keyof typeof eqs]
 
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    return eq === undefined || eq.equals(xv as any, yv as any)
+    return (
+      eq === undefined ||
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (eq as unknown as Eq<Value<A>>).equals(xv as any, yv as any)
```

Comment:
Do you know how `eq` gets its type `const eq: Eqs<A>[keyof Eqs<A>] & ({} | null)` after the equality test with undefined while it was simply `const eq: Eqs<A>[keyof Eqs<A>]` before it?

Reply:
I'm not seeing that inferred intersection. I wouldn't waste much time on it though, these libraries are all absolutely littered with assertions. I don't think there's much we can do about it (though you're welcome to try 😉). The only real concern for me is what's consumer-facing.